plsync/mlabconfig_test.py

```diff
@@ -153,27 +158,29 @@ def test_export_experiment_records_flattened(self):
                                    use_initscript=True,
                                    ipv6='all')]
         expected_results = [
-            mlabconfig.format_a_record('foo-abc-abc01', '192.168.1.11'),
             mlabconfig.format_a_record('foo-abc-mlab2-abc01', '192.168.1.24'),
-            mlabconfig.format_a_record('foo-abcv4-abc01', '192.168.1.11'),
             mlabconfig.format_a_record('foo-abc-mlab2v4-abc01', '192.168.1.24'),
-            mlabconfig.format_aaaa_record('foo-abc-abc01',
+            mlabconfig.format_aaaa_record('foo-abc-mlab3-abc01',
+                                          '2400:1002:4008::37'),
+            mlabconfig.format_aaaa_record('foo-abc-mlab1v6-abc01',
                                           '2400:1002:4008::11'),
+        ]
+
+        unexpected_results = [
+            mlabconfig.format_a_record('foo-abc-abc01', '192.168.1.24'),
+            mlabconfig.format_a_record('foo-abcv4-abc01', '192.168.1.24'),
             mlabconfig.format_aaaa_record('foo-abc-abc01',
                                           '2400:1002:4008::37'),
-            mlabconfig.format_aaaa_record('foo-abc-mlab3-abc01',
-                                          '2400:1002:4008::37'),
             mlabconfig.format_aaaa_record('foo-abcv6-abc01',
                                           '2400:1002:4008::11'),
-            mlabconfig.format_aaaa_record('foo-abc-mlab1v6-abc01',
-                                          '2400:1002:4008::11'),
         ]
 
         mlabconfig.SSL_EXPERIMENTS = ['abc_foo']
         mlabconfig.export_experiment_records(output, self.sites, experiments)
 
         results = output.getvalue().split('\n')
         self.assertContainsItems(results, expected_results)
+        self.assertDoesNotContainsItems(results, unexpected_results)
```

Comment:
I suspect you added the custom function `assertContainsItems()` because of the size of the output list. We could certainly create an `expected_results` list that matched the output, but it would be very unwieldy. This is probably a trade-off between readability of the unittest code and comprehensiveness of the test. I think I'm personally in favor of keeping the current system and just adding a comment as to why those custom functions exists.

To back up my thoughts on this, here is the output of `export_experiment_records()`:

```
[
  '',
  '; foo.abc v4',
  'foo.abc.abc01                     IN  A   \t192.168.1.11',
  'foo.abc.mlab1.abc01               IN  A   \t192.168.1.11',
  'foo.abc.abc01                     IN  A   \t192.168.1.24',
  'foo.abc.mlab2.abc01               IN  A   \t192.168.1.24',
  'foo.abc.abc01                     IN  A   \t192.168.1.37',
  'foo.abc.mlab3.abc01               IN  A   \t192.168.1.37',
  '',
  '; foo.abc v4 decorated',
  'foo.abcv4.abc01                   IN  A   \t192.168.1.11',
  'foo.abc.mlab1v4.abc01             IN  A   \t192.168.1.11',
  'foo.abcv4.abc01                   IN  A   \t192.168.1.24',
  'foo.abc.mlab2v4.abc01             IN  A   \t192.168.1.24',
  'foo.abcv4.abc01                   IN  A   \t192.168.1.37',
  'foo.abc.mlab3v4.abc01             IN  A   \t192.168.1.37',
  '',
  '; foo.abc v6',
  'foo.abc.abc01                     IN  AAAA\t2400:1002:4008::11',
  'foo.abc.mlab1.abc01               IN  AAAA\t2400:1002:4008::11',
  'foo.abc.abc01                     IN  AAAA\t2400:1002:4008::24',
  'foo.abc.mlab2.abc01               IN  AAAA\t2400:1002:4008::24',
  'foo.abc.abc01                     IN  AAAA\t2400:1002:4008::37',
  'foo.abc.mlab3.abc01               IN  AAAA\t2400:1002:4008::37',
  '',
  '; foo.abc v6 decorated',
  'foo.abcv6.abc01                   IN  AAAA\t2400:1002:4008::11',
  'foo.abc.mlab1v6.abc01             IN  AAAA\t2400:1002:4008::11',
  'foo.abcv6.abc01                   IN  AAAA\t2400:1002:4008::24',
  'foo.abc.mlab2v6.abc01             IN  AAAA\t2400:1002:4008::24',
  'foo.abcv6.abc01                   IN  AAAA\t2400:1002:4008::37',
  'foo.abc.mlab3v6.abc01             IN  AAAA\t2400:1002:4008::37',
  '',
  '; foo.abc v4 flattened',
  'foo-abc-mlab1-abc01               IN  A   \t192.168.1.11',
  'foo-abc-mlab2-abc01               IN  A   \t192.168.1.24',
  'foo-abc-mlab3-abc01               IN  A   \t192.168.1.37',
  '',
  '; foo.abc v4 decorated flattened',
  'foo-abc-mlab1v4-abc01             IN  A   \t192.168.1.11',
  'foo-abc-mlab2v4-abc01             IN  A   \t192.168.1.24',
  'foo-abc-mlab3v4-abc01             IN  A   \t192.168.1.37',
  '',
  '; foo.abc v6 flattened',
  'foo-abc-mlab1-abc01               IN  AAAA\t2400:1002:4008::11',
  'foo-abc-mlab2-abc01               IN  AAAA\t2400:1002:4008::24',
  'foo-abc-mlab3-abc01               IN  AAAA\t2400:1002:4008::37',
  '',
  '; foo.abc v6 decorated flattened',
  'foo-abc-mlab1v6-abc01             IN  AAAA\t2400:1002:4008::11',
  'foo-abc-mlab2v6-abc01             IN  AAAA\t2400:1002:4008::24',
  'foo-abc-mlab3v6-abc01             IN  AAAA\t2400:1002:4008::37',
  ''
]
```

Do you agree with my assessment, and following your suggestion of just adding helping comments?